managers/asset-operations-manager.js

```diff
@@ -370,18 +375,26 @@ class AssetOperationsManager {
                 }
 
                 publicAssertionId = assertionIds[state];
+
+                if (state === assertionIds.length - 1) stateFinalized = true;
             } else if (assertionIds.includes(state)) {
                 publicAssertionId = state;
+
+                if (state === assertionIds[assertionIds.length - 1]) stateFinalized = true;
             } else if (/^0x[a-fA-F0-9]{64}$/.test(state)) {
                 const unfinalizedState = await this.blockchainService.getUnfinalizedState(
                     tokenId,
                     blockchain,
                 );
-    
-                if (unfinalizedState != null && unfinalizedState !== ZeroHash && state === unfinalizedState) {
+
+                if (
+                    unfinalizedState != null &&
+                    unfinalizedState !== ZeroHash &&
+                    state === unfinalizedState
+                ) {
                     publicAssertionId = unfinalizedState;
```

Comment:
In this case state is unfinalized

Reply:
Again, my bad. Since it's false by default it's not really necessary to mark it false here

Reply:
I'd say it's better to leave it like this so the code is easier to read, but welcome to hear other team members thoughts on this :D